app/controllers/authentications_controller.rb

```diff
@@ -96,4 +97,14 @@ def redirect_if_current_user_verified
     return if current_user.nil?
     redirect_to root_path if current_user.access.mobile_or_oauth_verified?
   end
+
+  def redirect_invalid_github_account
+    return if github_api.created_at < 1.month.ago && github_api.repository_has_language?
```

Comment:
Don't we want this check to be for no repositories?

`return if github_api.created_at < 1.month.ago && !github_api.repository_has_language?`

Reply:
The `created_at < 1.month.ago` comparision can be a little confusing due to how datetime comparision works. If I didn't have tests, I wouldn't have been confident about this code.

This line can be translated as:
```ruby
return if account_is_older_than_a_month? && account_has_a_repository_with_language?
redirect_to path, message: 'This account does not fulfill our requirements'
```
Does it feel correct now?

Reply:
Yes, these are both positive tests: the account must be older than 1 month and there is a public repository with recognizable code.  Thanks for the explanation,